src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -554,7 +538,10 @@ private ImmutableArray<SymbolDeclaredCompilationEvent> GetPendingSymbolDeclaredE
         /// Returns false if the event has already been processed for the analyzer OR is currently being processed by another task.
         /// If true, then it returns a non-null <paramref name="state"/> representing partial analysis state for the given event for the given analyzer.
         /// </returns>
-        public bool TryStartProcessingEvent(CompilationEvent compilationEvent, DiagnosticAnalyzer analyzer, out AnalyzerStateData state)
+        public bool TryStartProcessingEvent(
+            CompilationEvent compilationEvent,
+            DiagnosticAnalyzer analyzer,
+            [NotNullWhen(returnValue: true)] out AnalyzerStateData? state)
```

Comment:
>returnValue [](start = 25, length = 11)

nit: the attribute is common enough that we probably don't need to name the argument (also below)

Reply:
Sure, and with @akhera99's awesome work on inline parameter hints, there is no reason to have it now! I will remove.